diff --git a/configure b/configure
index e974d1f..ce8829f 100755
--- a/configure
+++ b/configure
@@ -115,6 +115,8 @@ case "$1" in
       echo '    [--static] [--64] [--libdir=LIBDIR] [--sharedlibdir=LIBDIR]' | tee -a configure.log
       echo '    [--includedir=INCLUDEDIR] [--archs="-arch i386 -arch x86_64"]' | tee -a configure.log
         exit 0 ;;
+    CC=*) CC=`echo $1 | sed 's/.*=//'`; old_cc="$CC" ; shift ;;
+    CFLAGS=*) CFLAGS=`echo $1 | sed 's/.*=//'`; old_cflags="$CFLAGS" ; shift ;;
     -p*=* | --prefix=*) prefix=`echo $1 | sed 's/.*=//'`; shift ;;
     -e*=* | --eprefix=*) exec_prefix=`echo $1 | sed 's/.*=//'`; shift ;;
     -l*=* | --libdir=*) libdir=`echo $1 | sed 's/.*=//'`; shift ;;
@@ -138,9 +140,9 @@ case "$1" in
     -w* | --warn) warn=1; shift ;;
     -d* | --debug) debug=1; shift ;;
     *)
-      echo "unknown option: $1" | tee -a configure.log
+      echo "Warning: unknown option: $1" | tee -a configure.log
       echo "$0 --help for help" | tee -a configure.log
-      leave 1;;
+      shift ;;
     esac
 done
 
